plugins/aws/fix_plugin_aws/collector.py

```diff
@@ -276,6 +288,25 @@ def work_done(_: Future[None]) -> None:
         when_done.add_done_callback(work_done)
         return when_done
 
+    def collect_usage_metrics(self, resources: List[Type[AwsResource]], builder: GraphBuilder) -> None:
+        futures: List[Future[Tuple[List[cloudwatch.AwsCloudwatchQuery], Dict[str, AwsResource], Dict[str, Any]]]] = [
+            builder.submit_work("cloudwatch", resource.collect_usage_metrics, builder) for resource in resources
+        ]
+        usage_metric_data = [future.result() for future in as_completed(futures)]
+        for data in usage_metric_data:
```

Comment:
No need to execute this in a future - only queries are created without performing any IO.
Instead of creating a list all queries, consider iterating the list until the next chunk is full (500 elements) and then you could spawn a future here.